libcalico-go/lib/clientv3/ippool.go

```diff
@@ -283,7 +283,8 @@ func convertIpPoolFromStorage(pool *apiv3.IPPool) error {
 		pool.Spec.VXLANMode = apiv3.VXLANModeNever
 	}
 
-	if pool.Spec.AssignmentMode == "" {
+	assignmentMode := pool.Spec.AssignmentMode
+	if &assignmentMode == nil {
```

Comment:
I think something is off here - it doesn't make sense to check if the address of a variable is `nil`.